05/app.js

```diff
@@ -0,0 +1,37 @@
+function Student(initFirstName, initLastName) {
+    this.firstName = initFirstName;
+    this.lastName = initLastName;
+    this.grades = {};
+    this.allGrades = [];
+}
+
+Student.prototype.addGrade = function (subject, grade) {
+    if (typeof this.grades[subject] === 'undefined') {
+        this.grades[subject] = [];
+        console.log('Dodano przedmiot ' + subject + ' oraz ocenę: ' + grade);
+    }
+
+    this.grades[subject].push(grade);
+    this.allGrades.push(grade);
+    console.log('Dodano ocenę ' + grade + ' do przedmiotu ' + subject);
+}
+
+Student.prototype.getAverage = function (subject) {
+    let sum = 0;
+    let avg = 0;
+    if (typeof this.grades[subject] === 'undefined') {
+        this.allGrades.forEach(function (element) {
+            sum += element;
+        })
+        avg = sum / this.allGrades.length
+        console.log('Suma wszystkich ocen tego studenta to: ' + avg);
```

Comment:
To rozwiązanie nie jest złe. Super, że znalazłeś alternatywę.
Jednak mógłbyś to zrobić tak:
```
let all = [];
for(const subject in this.grades) {
    all = all.concat(this.grades[subject])
}
```
Teraz masz już wszystkie liczby w 1 tablicy.
Opcji jest bardzo dużo, część z nich jeszcze będziemy omawiać. Zobacz tutaj:
https://stackoverflow.com/questions/4156101/copy-array-items-into-another-array

Reply:
Super, bardzo dziękuję. Takiego rozwiązania poszukiwałem :)